### PR TITLE
box_cheatsheet

### PR DESCRIPTION
**Pull request by kannannbabu**
DDG Community Kochi
#2127

### CODE REVIEW

**Comment by daxtheduck** [comment]
@kannannbabu Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/box_cheatsheet.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "id": ".box_cheat_sheet",
+  "name": ".box_cheat_sheet",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/box_cheatsheet.json]: You sure you want that period?

File: share/goodie/cheat_sheets/json/box_cheatsheet.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  "description": "When you’re learning Scala, one of the first concepts that clicks is pattern matching. Once you get the “hammer” of pattern matching in your Scala toolbox, everything is a nail; you want to use it everywhere for everything. Pattern matching is very powerful and easy to use, but it turns out for many, many simple cases using a higher-order function is more concise and performant. I have Tony Morris’s Option Cheat Sheet .",
+  "metadata": {
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/box_cheatsheet.json]: The description can be much shorter! Only needs to outline what the cheat sheet does, see the Vim example in [the docs](http://docs.duckduckhack.com/frontend-reference/cheat-sheet-reference.html).

File: share/goodie/cheat_sheets/json/box_cheatsheet.json (hunk 1 above)
Comment (MariagraziaAlastra) [inline share/goodie/cheat_sheets/json/box_cheatsheet.json]: @kannannbabu A good name could be "Box Cheat Sheet" :)

**Comment by MariagraziaAlastra** [comment]
Hey @kannannbabu, thanks for contributing!
Could you please create an IA Page [here](https://duck.co/ia/new_ia) and then paste the link to it in your first comment? 
You can find more info [in the docs](http://docs.duckduckhack.com/submitting/pull-request.html#submitting-a-pull-request).
Thanks!

**Comment by moollaza** [comment]
@kannannbabu thanks again for all the hard work you've put into creating this Box Cheat Sheet. You've done a great job so far! We at DuckDuckGo, and the DuckDuckHack community really appreciate you taking the time to contribute.

It looks like this still needs some work?  If you're stuck or need any help, I'd love to chat if you're around this week or next. Please message me on slack (@moollaza) or send me an email (moollaza@duckduckgo.com) if you'd be interested in talking. 

Thanks again!

**Comment by jdorweiler** [comment]
This pull request is being closed due to inactivity but, if anyone would like to continue it, please @ mention 'duckduckhack-contributors' for help and http://docs.duckduckhack.com/submitting/submitting-overview.html for reference.

**Pull request closed without merging**